File: backend/src/main/java/com/example/beyazesya/service/SaleService.java
package com.example.beyazesya.service;

import com.example.beyazesya.dto.AddSaleDetailRequest;
import com.example.beyazesya.dto.CreateSaleRequest;
import com.example.beyazesya.dto.SaleReportResponse;
import com.example.beyazesya.dto.SaleResponse;
import com.example.beyazesya.dto.UpdateSaleDetailQuantityRequest;
import com.example.beyazesya.dto.UpdateSaleStatusRequest;
import java.util.List;

public interface SaleService {

    SaleResponse createSale(CreateSaleRequest request);

    SaleResponse addSaleDetail(Integer saleId, AddSaleDetailRequest request);

    SaleResponse updateSaleDetailQuantity(
            Integer saleId,
            Integer productId,
            UpdateSaleDetailQuantityRequest request
    );

    SaleResponse updateSaleStatus(Integer saleId, UpdateSaleStatusRequest request);

    List<SaleReportResponse> getCustomerSales(Integer customerId);

    List<SaleReportResponse> getSaleReport();
}

File: frontend/src/services/saleService.js
import { get, patch, post, put } from './apiClient'

export function createSale(payload) {
  return post('/api/sales', payload)
}

export function addSaleDetail(saleId, payload) {
  return post(`/api/sales/${saleId}/details`, payload)
}

export function updateSaleDetailQuantity(saleId, productId, payload) {
  return put(`/api/sales/${saleId}/details/${productId}`, payload)
}

export function updateSaleStatus(saleId, payload) {
  return patch(`/api/sales/${saleId}/status`, payload)
}

export function getSaleReport() {
  return get('/api/sales/report')
}

export function getCustomerSales(customerId) {
  return get(`/api/sales/customer/${customerId}`)
}
